README.md

```diff
@@ -49,10 +49,12 @@ The Make and CMake build systems create three libraries: libmbedcrypto, libmbedx
 
 You need the following tools to build the library with the provided makefiles:
 
-* GNU Make or a build tool that CMake supports.
-* A C99 toolchain (compiler, linker, archiver). We actively test with GCC 5.4, Clang 3.8, IAR8 and Visual Studio 2013. More recent versions should work. Slightly older versions may work.
-* Python 3 to generate the test code, and to generate sample programs in the development branch.
+* GNU Make 3.82 or a build tool that CMake supports.
+* A C99 toolchain (compiler, linker, archiver). We actively test with GCC 5.4, Clang 6, IAR8 and Visual Studio 2013. More recent versions should work. Slightly older versions may work.
+* Python 3.6 to generate the test code, and to generate sample programs in the development branch.
 * Perl to run the tests, and to generate some source files in the development branch.
+* CMake 3.10.2 or later (if using CMake).
```

Comment:
This excludes SLES 12 (CMake 3.5.2). Is there significant benefit in requiring 3.10?

Reply:
SLES 12 is 7 years old (they skipped 13 and 14, so 15 was next), so unlikely to affect too many users. I discussed this one with @gyuri-szing who didn't have a very strong view either way but did point out that it's easy to get a later version for SLES 12 users, so overall I'm inclined to go with something more recent - we've recently seen difficulties caused by specifying too-old versions of tools, but problems the other way seem less likely, and easier to solve.

Reply:
We do have users who care. From https://lists.trustedfirmware.org/pipermail/mbed-tls/2020-April/000066.html :

> we support RHEL6 until it's retirement (30 November 2020).
> (…)
> Yes, we support all Ubuntu LTS versions until their standard support
ends (April 2021 for 16.04).
> (…)
> All Debian and SLES versions that are still in regular maintance (ie,
not including any "extended security maintenance" support).
> It would be nice if it's at least possible to build mbedtls on/for these
platforms until they go out of regular support. Preferably with their
default packages compilers.

Not an explicit request for cmake, but since cmake is reasonably good at having features that are optionally supported, it would be good if at least basic operation worked with older versions. Ideally this would apply to RHEL7 but its cmake is so old (2.8.12.2) that I think that's just not practical.

Reply:
I'm not a fan of requiring CMake 3.10 for basic CMake operation, but not strongly against either, so I'm not rejecting this change.

Reply:
I think the current wording doesn't prevent us from adding a compatibility mode (supporting basic operations but perhaps not everything) later if users of SLES 12 come complaining on the issue tracker or mailing list. But it gives us a bit more freedom if nobody complains, which I think is good.